st2common/st2common/config.py

```diff
@@ -500,6 +500,28 @@ def register_opts(ignore_errors=False):
         dispatcher_pool_opts, group="actionrunner", ignore_errors=ignore_errors
     )
 
+    graceful_shutdown_opts = [
+        cfg.BoolOpt(
+            "graceful_shutdown",
+            default=False,
+            help="This will enable the graceful shutdown and wait for ongoing requests to complete until exit_timeout.",
+        ),
+        cfg.IntOpt(
+            "exit_timeout",
+            default=300,
+            help="How long to wait for process (in seconds) to exit after receiving shutdown signal.",
+        ),
+        cfg.IntOpt(
+            "sleep_delay",
+            default=2,
+            help="Time interval between subsequent queries to check running executions.",
```

Comment:
I think this option needs a more descriptive name. Maybe something with `interval` like `exit_still_active_check_interval` or `ready_to_exit_check_interval`?